LocalKeyClientEncryption/customer_provided_cse_migration.py

```diff
@@ -0,0 +1,122 @@
+################################################################
+# this application goes through the process of migration from   #
+# private client side encryption to server side encryption     #
+# this application includes three types of server side         #
+# encryption: Microsoft managed keys through encryption        #
+# scope, customer managed keys through encryption scope,       #
+# and customer provided keys. This application requires that   #
+# environmental variables, keys, and encryption scopes are set,#
+# as well as private encryption keys, container, and blob names#
+################################################################
+import os
+import uuid
+import privateConfig
+from PIL import Image
+from azure.storage.blob import BlobServiceClient, BlobClient, ContainerClient, ContainerEncryptionScope, \
+    CustomerProvidedEncryptionKey
+from azure.identity import ClientSecretCredential
+from azure.keyvault.keys import KeyClient
+from cryptography.fernet import Fernet  # python extension for decryption
+
+
+def download_blob(filename, bs_client, cont_name):
+    # download encrypted blob from azure storage
+    # access specific container and blob to download from using blob client
+    blob_client = bs_client.get_blob_client(container=cont_name, blob=filename)
+    print("\nDownloading blob from Azure storage...")
+    # write encrypted contents of blob to a file
+    with open(filename, "wb+") as download_file:
+        download_file.write(blob_client.download_blob().readall())
+
+def make_key(key_client):

```

Comment:
if the user is using keyvault based encryption scope, that means the customer wants to use their own key. you don't need to make key for them right? You can take encryption scope as a black box, if you want to tell service to do encryption, just give it the black box name. Inside the black box, key would be fetched and used for service side encryption. Only when user wants to touch the key in black box, they can put their desired key in keyvault, then service would know to use that key for encryption.